webapp/public/multi.html

```diff
@@ -53,7 +53,7 @@
             var divider = document.getElementById("divider");
             var localhost = window.location.hostname == "localhost";
             var editor = (pxtConfig ? pxtConfig.relprefix : '/').replace(/-*$/, '');
-            var flags = "?nestededitorsim=1&editorlayout=ide&nosandbox=1";
+            var flags = "?nestededitorsim=1&editorlayout=ide&nosandbox=1&parentOrigin=" + window.location.origin;
```

Comment:
I noticed that sometimes parentOrigin is a parameter and sometimes you just grab the location.origin

Is there a quick description for why these can differ?

Reply:
It's the `parentOrigin` from the perspective of the iframe, which is the location of the current window spawning said iframe. 

There are three origins that I care about in the editor. There is the origin of my parent window, my current origin, and the origins of any simulators the editor has spawned. 
#### Posting Messages
Since the `SimulatorDriver` can send broadcast messages to its parent window, it needs to know its parent origin. Sending messages to simulator iframes will always use the `simUrl`, and broadcasts to dependent editors will use the same origin as the current editor (working on this now). 
#### Receiving Messages
On the receiving end, messages can come from any of those three origins I mentioned before. In that case, the message just has to come from one of them to not be discarded.

Please let me know if this has answered your question.